PolyEngine/RenderingDevice/OpenGL/Src/BlinnPhongRenderingPass.cpp

```diff
@@ -19,10 +19,16 @@ const size_t MAX_LIGHT_COUNT_DIRECTIONAL = 8;
 BlinnPhongRenderingPass::BlinnPhongRenderingPass()
 : RenderingPassBase("Shaders/blinn-phongVert.shader", "Shaders/blinn-phongFrag.shader")
 {
-	GetProgram().RegisterUniform("float", "uSpecularStrength");
+	GetProgram().RegisterUniform("vec4", "uCameraPosition");
+	GetProgram().RegisterUniform("vec4", "uCameraForward");
 
-	GetProgram().RegisterUniform("vec4", "uAmbientLight.Base.Color");
-	GetProgram().RegisterUniform("float", "uAmbientLight.Base.Intensity");
+	GetProgram().RegisterUniform("vec4", "uMaterial.Ambient");
+	GetProgram().RegisterUniform("vec4", "uMaterial.Diffuse");
+	GetProgram().RegisterUniform("vec4", "uMaterial.Specular");
+	GetProgram().RegisterUniform("float", "uMaterial.Shininess");
+
+	GetProgram().RegisterUniform("vec4", "uAmbientLight.Color");
+	GetProgram().RegisterUniform("float", "uAmbientLight.Intensity");
```

Comment:
We need to start thinking about adding UBO support.

Reply:
Will start think about UBO, but I would do Forward+ first to have clear picture on which uniforms shading do exactly need.

Reply:
UBO from Unreal Engine 4:
https://docs.unrealengine.com/latest/INT/API/Runtime/Engine/FViewUniformShaderParameters/index.html